### ISSUE TITLE
[java] LambdaCanBeMethodReference - false-positive on overloaded methos

### ISSUE THREAD

**Issue by Wolf2323** [pmd/pmd#5043]
**Affects PMD Version:** 7.1.0

**Rule:** LambdaCanBeMethodReference

Please provide the rule name and a link to the rule documentation:
https://pmd.github.io/pmd/pmd_rules_java_codestyle.html#lambdacanbemethodreference

**Description:**
When a method is overloaded like this
```java
    public void closeInventory();
    public void closeInventory(@NotNull InventoryCloseEvent.Reason reason);
```
it is not possible to use a method reference, but the rule report it

**Code Sample demonstrating the issue:**

```java
            Bukkit.getScheduler().runTask(BetonQuest.getInstance(), () -> player.closeInventory());
```

**Expected outcome:**
no violation

[INFO] PMD Failure: org.betonquest.betonquest.conversation.InventoryConvIO:392 Rule:LambdaCanBeMethodReference Priority:3 Lambda expression could be written as a method reference: `player::closeInventory`.

**Running PMD through:** *Maven*

**Comment by cowwoc** [comment]
For what it's worth, I can reproduce this problem when a single (non-overloaded) method invokes an overloaded method.

My method `public ValidationResult test(Predicate<T> condition)` is passed `value -> value.isNaN()` where `value` is a `Double`.

I get: `Rule:LambdaCanBeMethodReference Priority:3 Lambda expression could be written as a method reference: `Double::isNaN`.` but the compiler disagrees: `Reference to 'isNaN' is ambiguous, both 'isNaN(double)' and 'isNaN()' match`.

**Comment by oowekyala** [comment]
Interesting, thanks for your input

**Comment by hgschmie** [comment]
Here is a much simpler example:

```java
public void foo() {
        Function<Integer, String> x = v -> Integer.toString(v);

        System.out.println(x.apply(100));
    }
```

reports

```
[WARNING] PMD Failure: <line that represents the definition of x> Rule:LambdaCanBeMethodReference Priority:3 Lambda expression could be written as a method reference: `Integer::toString`.
```

but rewriting to `Function<Integer, String> x = Integer::toString;`

leads to a red squiggle line in IntelliJ (uh, oh!) and

```
Compilation failure
[ERROR] <line that represents the definition of x>: incompatible types: invalid method reference
[ERROR]     reference to toString is ambiguous
[ERROR]       both method toString(int) in java.lang.Integer and method toString() in java.lang.Integer match
```
